src/components/molecules/DateInput/YearBar/YearBar.tsx

```diff
@@ -0,0 +1,256 @@
+    }
+  };
+  const onMouseDown = (e: React.MouseEvent<HTMLDivElement>) => {
+    setIsMouseDown(true);
+    clearScrollParameters();
+    setScrollingState(false);
+    clearInterval(timeInterval);
+    clearInterval(accelerateInterval);
+    if (wrapperRef.current) {
+      setStartX(e.pageX - wrapperRef.current.offsetLeft);
+      setScrollPosition(wrapperRef.current.scrollLeft);
+    }
+  };
+
+  const onMouseUp = () => {
+    setIsMouseDown(false);
+    if (!isMobileDevice() && (movementX >= 0.66 || movementX <= -0.66)) {
+      accelerateScroll();
+      setScrollingState(true);
+      setTimeInterval(
+        setInterval(() => {
+          setScrollTime((prev) => prev + 15);
+        }, 1)
+      );
+    } else {
+      handleScrolling();
+    }
+  };
+  const onTouchEnd = () => {
+    setIsMouseDown(false);
+    setScrollingState(true);
+    setTimeInterval(
+      setInterval(() => {
+        setScrollTime((prev) => prev + 15);
+      }, 1)
+    );
+  };
+
+  const onScroll = () => {
+    if (!isMouseDown) {
+      setScrollPosition(+`${wrapperRef.current?.scrollLeft}`);
+      manageCurrentYear();
+    }
+  };
+
+  const manageScrollVelocity = useCallback(() => {
+    setScrollVelocity(+(Constants.scrollRoad / scrollTime).toFixed(5));
+  }, [scrollTime]);
+
+  useEffect(() => {
+    setIsFirstRender(false);
+  }, [setIsFirstRender, isFirstRender]);
+
+  useEffect(() => {
+    if (scrollingState) {
+      setSelectedYear(currentYear);
+    }
+  }, [scrollingState, setSelectedYear, currentYear]);
+
+  useEffect(() => {
+    if (!scrollingState) {
+      setCurrentYear(year);
+      scrollToCurrentYear(wrapperRef, +`${getBoxPosition(year)}`, 'smooth');
+    }
+  }, [year, getBoxPosition, scrollingState]);
+
+  useEffect(() => {
+    handleRemoveClass(wrapperRef);
+    handleAddClass(wrapperRef, currentYear);
+  }, [wrapperRef, currentYear, handleRemoveClass, handleAddClass]);
+
+  useEffect(() => {
+    manageScrollVelocity();
+  }, [scrollTime, manageScrollVelocity]);
+
+  useEffect(() => {
+    if (scrollingState && scrollVelocity < Constants.scrollingBorder) {
+      // 3
+      handleScrolling();
+    }
+  }, [handleScrolling, scrollVelocity, scrollingState]);
+
+  useEffect(() => {
+    // 1
+    if (scrollVelocity < Constants.clearBorder && scrollTime > 1000) {
+      setScrollingState(false);
+      clearTimeInterval();
+      clearAccelerateInterval();
+    }
+  }, [scrollVelocity, scrollTime, clearAccelerateInterval, clearTimeInterval]);
+
+  useEffect(() => {
+    if (isFirstRender) {
+      scrollToCurrentYear(wrapperRef, +`${getBoxPosition(currentYear)}`, 'auto');
+    }
+  }, [isFirstRender, wrapperRef, currentYear, getBoxPosition]);
+
+  return (
+    <StyledYearBar>
+      <StyledGradient className="left" />
+      <StyledWrapper onScroll={onScroll} className={isMouseDown ? 'active' : ''} ref={wrapperRef}>
+        {boxes.map(({ value }) => (
+          <StyledYear
+            onTouchStart={(e) => onTouchStart(e)}
```

Comment:
We can use shortcut here
`onTouchStart={(e) => onTouchStart(e)}` into `onTouchStart={onTouchStart}`